iceoryx_posh/source/roudi/roudi_lock.cpp

```diff
@@ -13,22 +13,22 @@
 // limitations under the License.
 
 #include "iceoryx_posh/internal/roudi/roudi_lock.hpp"
+#include "iceoryx_utils/platform/socket.hpp"
+#include "iceoryx_utils/platform/unistd.hpp"
 
-#include <arpa/inet.h>
 #include <stdlib.h>
-#include <sys/socket.h>
-#include <sys/types.h>
-#include <unistd.h>
 
 namespace iox
 {
 namespace roudi
 {
 RouDiLock::RouDiLock()
+    : m_socket_fd(cxx::makeSmartC(socket, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {-1}, {}, AF_INET, SOCK_STREAM, 0)
```

Comment:
Once the unix domain socket wrapper is merged, we could think of using this one here and not directly the socket calls. 
What do you think?

Reply:
You are completely right. I will use your abstraction as soon as it is available ...